File: Aula_Ricardo/Programação_orientada_a_objetos/java/Metodos_com_nomes_diferentes.java
public class Calculadora {
    
    public int somar(int a, int b) {
        return a + b;
    }

    public int somar(int a, int b, int c) {
        return a + b + c;
    }

    public static void main(String[] args) {
        Calculadora calc = new Calculadora();

        System.out.println(calc.somar(1, 2)); 

        System.out.println(calc.somar(1, 2, 3)); 
    }
}

File: Aula_Ricardo/Programação_orientada_a_objetos/python/Metodos_com_nomes_diferentes.py
class Calculadora: 
    def somar(self,a, b, c= None):
        if c is not None:
            return a + b + c
        return a + b

calc = Calculadora()
print(calc.somar(1, 2))
print(calc.somar(1, 2, 3))